### PR TITLE
Add telemetry markers for when user is taken from app to AAD login screen

### PR DESCRIPTION
**Pull request by anthony-ngu**
The work done here adds a 'reason' attribute to telemetry currently being gathered for adal login calls

### CODE REVIEW

File: lib/adal-angular.js [hunk 1]
```diff
@@ -225,7 +226,7 @@ if (typeof module !== 'undefined' && module.exports) {
                             _adal._saveItem(_adal.CONSTANTS.STORAGE.START_PAGE_PARAMS, JSON.stringify(toParams));
 
                             _adal.info('State change event for:' + $location.$$url);
-                            loginHandler();
+                            loginHandler('adal:unauthenticated&renew_inactive');
```
Comment (amitapl) [inline lib/adal-angular.js]: this is the same string as another, please diffrentiate it a bit so we can tell which is which

File: lib/adal.js [hunk 2]
```diff
@@ -165,7 +165,9 @@ AuthenticationContext = function (config) {
  * Gets initial Idtoken for the app backend
  * Saves the resulting Idtoken in localStorage.
  */
-AuthenticationContext.prototype.login = function () {
+AuthenticationContext.prototype.login = function (loginReason) {
+    loginReason = loginReason || '';
```
Comment (amitapl) [inline lib/adal.js:169]: is this actually required, it can be null right?

Comment (anthony-ngu) [reply lib/adal.js:169]: yes it can be null coming in, but if it is shouldn't we provide an empty string for telemetry purposes?

Comment (amitapl) [reply lib/adal.js:169]: you can but you don't have to

**Pull request merged**